core/includes/bootstrap.inc

```diff
@@ -1011,6 +1016,7 @@ function backdrop_get_filename($type, $name, $filename = NULL) {
  * @deprecated since 1.0
  */
 function variable_initialize($conf = array()) {
+  watchdog_deprecated_function('bootstrap', __FUNCTION__);
```

Comment:
I didn't remember seeing it when I was testing the code, so I took another look and found that call doesn't generate a watchdog message.  The function `module_implements()` doesn't exist that early on, so `watchdog()` drops the message.

`variable_initialize()` is called during bootstrap and in backdrop_web_test_case.php.  A quick google search didn't find any Drupal contributed modules that use it.  

I have no objection to replacing that `watchdog_deprecated_function()` call with a comment.